### PR TITLE
Handle rejected pushes gracefully

### PR DESCRIPTION
**Pull request by dbrgn**
If a push was rejected with a HTTP 400 response (which can indicate an
expired push token), don't show a protocol error. Instead, offer to
forget & recreate the session.

Before this PR, a protocol error was shown.

### CODE REVIEW

File: src/partials/messenger.ts [hunk 1]
```diff
@@ -165,6 +165,40 @@ export class DeviceUnreachableController extends DialogController {
...
+    }
+
+    /**
+     * Remove the stored session.
+     */
+    public forget(): void {
+        this.log.info('Forgetting stored session');
+        this.trustedKeyStore.clearTrustedKey();
+        this.cancel();
+        this.$window.location.reload();
```
Comment (dbrgn) [inline src/partials/messenger.ts]: Lazy solution to ensure that all data is gone.

File: src/partials/dialog.push-rejected.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<md-dialog aria-label="Device Unreachable">
```
Comment (lgrahl) [inline src/partials/dialog.push-rejected.html]: Probably copy & paste leftover?

File: src/services/push.ts [hunk 3]
```diff
@@ -124,6 +124,7 @@ export class PushSession {
      *
      * @throws TimeoutError in case the maximum amount of retries has been
      *   reached.
+     * @throws PushError if the push was rejected by the push relay server.
```
Comment (lgrahl) [inline src/services/push.ts]: IIRC `Error` is not being used any more

File: src/exceptions.ts [hunk 4]
```diff
@@ -16,3 +16,12 @@
  */
 
 export class TimeoutError extends Error {}
+
+export class PushError extends Error {
+    public readonly responseCode: number;
```
Comment (lgrahl) [inline src/exceptions.ts]: Nit: Would prefer `code` or `statusCode`.

File: src/services/push.ts (hunk 3 above)
Comment (dbrgn) [reply src/services/push.ts]: It is.

**Pull request merged**